src/Silicon.cpp

```diff
@@ -202,10 +504,10 @@ namespace galsim {
                 double charge = *ptr;
                 if (charge == 0.0) continue;
 
-                int polyi1 = std::max(i - _qDist, i1);
-                int polyi2 = std::min(i + _qDist, i2);
-                int polyj1 = std::max(j - _qDist, j1);
-                int polyj2 = std::min(j + _qDist, j2);
+                int polyi1 = std::max(i - (_qDist+1), i1);
+                int polyi2 = std::min(i + (_qDist+1), i2);
+                int polyj1 = std::max(j - (_qDist+1), j1);
+                int polyj2 = std::min(j + (_qDist+1), j2);
```

Comment:
This is because of the problem I mentioned a while back where the outermost pixels in the box being updated were having their boundaries updated, but the ones immediately outside the box were not having their corresponding boundaries updated, causing them to diverge. To fix this it's necessary to loop over an additional pixel on each side of the box to make sure everything stays consistent. I thought it would be better to leave `_qDist` as it was and make the change here, since `_qDist` is passed in from outside the class and it might get confusing if its value no longer matches the one passed into the constructor.